chia/wallet/wallet_state_manager.py

```diff
@@ -1014,6 +1014,8 @@ async def new_coin_state(
 
         assert len(local_records) == len(coin_states)
         for coin_state, local_record in zip(coin_states, local_records):
+            if peer.closed:
```

Comment:
looking at where coin_state is called it looks like it is expected to throw if it fails

we should add a test to see that the state recovers correctly when this happens

Reply:
Thanks, good point. I changed it to raise instead and i also reworked the PR a bit overall.